File: 2. Python/pset6/cash/cash.py
import cs50

while True:
    dollars = cs50.get_float("Change owed: ")
    if dollars >= 0:
        break

cents = round(dollars * 100)
tcoins = 0

while cents >= 25:
    cents -= 25
    tcoins += 1

while cents >= 10:
    cents -= 10
    tcoins += 1

while cents >= 5:
    cents -= 5
    tcoins += 1

while cents >= 1:
    cents -= 1
    tcoins += 1

print(tcoins)

File: 1. C/pset1/cash/cash.c
#include <stdio.h>
#include <math.h>
#include <cs50.h>

int main(void)
{
    float dollars; //amount of change
    int tcoins = 0; // total coins

    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0); //if the chage is <0, then the user will have to insert a new value

    int cents = round(dollars * 100);

    while (cents >= 25)
    {
        cents = cents - 25; //decrement the total amount owed
        tcoins ++; //increment the counter
    }

    //same thing for the other coins

    while (cents >= 10)
    {
        cents = cents - 10;
        tcoins ++;
    }

    while (cents >= 5)
    {
        cents = cents - 5;
        tcoins ++;
    }

    while (cents >= 1)
    {
        cents = cents - 1;
        tcoins ++;
    }

    printf("%i\n", tcoins);
}